src/main/java/duke/task/TaskList.java

```diff
@@ -0,0 +1,213 @@
+            if (inputArr.length == 1) {
+                throw new DukeException("       OOPS!!! The task number cannot be empty.");
+            } else if (isNotNumeric(inputArr[1])) {
+                throw new DukeException("       OOPS!!! The task number must be numeric.");
+            } else {
+                int i = Integer.parseInt(inputArr[1]) - 1;
+                if (i > (this.tasks.size() - 1) || i < 0) {
+                    throw new DukeException("       OOPS!!! The task number is out of range. "
+                            + "Please use \"list\" to see the list of tasks.");
+                } else {
+                    return "     Noted. I've removed this task:\n"
+                            + "       " + this.tasks.remove(i).toString() + "\n"
+                            + "     Now you have " + this.tasks.size() + " tasks in the list.\n";
+                }
+            }
+        } else {
+            throw new DukeException("       OOPS!!! The format should be "
+                    + "\"delete ##\" where ## is the task number.");
+        }
+    }
+
+    /**
+     * Finds all task in task list that contains a given keyword.
+     *
+     * @param fullCommand Command given by user.
+     * @return Output for GUI.
+     * @throws DukeException If user input format is wrong.
+     */
+    public String find(String fullCommand) throws DukeException {
+        String[] inputArr = fullCommand.split(" ");
+        if (inputArr.length < 3) {
+            if (inputArr.length == 1) {
+                throw new DukeException("       OOPS!!! The keyword cannot be empty.");
+            } else {
+                StringBuilder sb = new StringBuilder("     Here are the matching tasks in your list:\n");
+                ArrayList<Task> temp = new ArrayList<>();
+                for (Task currTask : tasks) {
+                    String currDescription = currTask.getDescription();
+                    if (currDescription.contains(inputArr[1])) {
+                        temp.add(currTask);
+                    }
+                }
+                for (int j = 0; j < temp.size(); j++) {
+                    Task tempTask = temp.get(j);
+                    String msg = "     " + (j + 1) + "." + tempTask.toString() + "\n";
+                    sb.append(msg);
+                }
+                return sb.toString();
+            }
+        } else {
+            throw new DukeException("       OOPS!!! The format should be "
+                    + "\"find ####\" where #### is the keyword to search.");
+        }
+    }
+
+    /**
+     * Adds a task to the TaskList.
+     *
+     * @param type        The type of task.
+     * @param fullCommand Command given by user.
+     * @return Output for GUI.
+     * @throws DukeException If user input format is wrong.
+     */
+    public String addTask(String type, String fullCommand) throws DukeException {
+        String[] inputArr = fullCommand.split(" ");
+        switch (type) {
```

Comment:
I like the use of switch statements, makes it easier to read than a bunch of if-else blocks!